lib/acpi/acpi.c

```diff
@@ -217,6 +217,9 @@ static ACPI_STATUS dev_resource_enum_callback(ACPI_HANDLE obj_handle, UINT32 lev
 	}
 
 	child_dev = (struct acpi_dev *)&acpi.child_dev[acpi.num_dev++];
+	if (!dev_info->HardwareId.String) {
+		goto exit;
+	}
```

Comment:
Why is the if-statement *after* the assignment to `child_dev`? Seems like it should be before it, since there is no use of the `child_dev` pointer in the exist code path.